### PR TITLE
Refactoring for eventhub data writer

### PR DESCRIPTION
**Pull request by yukuai518**
The refactoring covers

1. Rename EventhubBatchAccumulator to SequentialBasedBatchAccumulator<D> 
2. Rename EventhubBatch to BytesBoundBatch<D> 
3. Create BatchedEventhubDataWriter inheriting from BufferedAsyncDataWriter<D>.
4. Remove  EventhubWriterConfigurationKeys
5. Change names in test.

### CODE REVIEW

File: gobblin-core-base/src/main/java/gobblin/writer/SequentialAndTTLBasedBatchAccumulator.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+/**
+ * Sequential and TTL based accumulator
+ * A producer can add a record to this accumulator. It generates a batch on the first record arrival. All subsequent records
+ * are added to the same batch until a batch size limit is reached. {@link BufferedAsyncDataWriter} keeps iterating available
+ * batches from this accumulator, all completed batches (full sized) will be popped out one by one but an incomplete batch
+ * keeps in the deque until a TTL is expired.
+ */
+
+public abstract class SequentialAndTTLBasedBatchAccumulator<D> extends BatchAccumulator<D> {
```
Comment (zxcware) [inline gobblin-core-base/src/main/java/gobblin/writer/SequentialAndTTLBasedBatchAccumulator.java]: `SequentialAndTTLBasedBatchAccumulator` is too verbose. `SequentialBatchAccumulator` is good enough. Usually, asynchronization comes together with timeout.

File: gobblin-core-base/src/main/java/gobblin/writer/SizeBoundAndTTLBasedBatch.java [hunk 2]
```diff
@@ -17,27 +17,26 @@
...
 
 /**
- * The Eventhub Batch which internally saves each record
- * For now we are using LinkedList as our internal memory storage
+ * A batch which internally saves each record in memory with bounded size limit
+ * Also a TTL value is configured, so that an eviction policy can be applied form upper layer.
  */
 @Alpha
-public class EventhubBatch extends Batch<String>{
+public class SizeBoundAndTTLBasedBatch<D> extends Batch<D>{
```
Comment (zxcware) [inline gobblin-core-base/src/main/java/gobblin/writer/SizeBoundAndTTLBasedBatch.java]: `SizeBoundAndTTLBasedBatch` -> `BytesBoundedBatch`

File: gobblin-core-base/src/main/java/gobblin/writer/SequentialAndTTLBasedBatchAccumulator.java (hunk 1 above)
Comment (yukuai518) [inline gobblin-core-base/src/main/java/gobblin/writer/SequentialAndTTLBasedBatchAccumulator.java]: Done.

File: gobblin-core-base/src/main/java/gobblin/writer/SizeBoundAndTTLBasedBatch.java (hunk 2 above)
Comment (yukuai518) [inline gobblin-core-base/src/main/java/gobblin/writer/SizeBoundAndTTLBasedBatch.java]: Done

File: gobblin-core-base/src/main/java/gobblin/writer/BytesBoundedBatch.java [hunk 3]
```diff
@@ -48,52 +47,52 @@ public boolean isTTLExpire() {
     return (System.currentTimeMillis() - creationTimestamp) >= ttlInMilliSeconds;
   }
 
-  private long getInternalSize(String record) {
-    return record.length() + this.OVERHEAD_SIZE_IN_BYTES;
+  private long getInternalSize(D record) {
+    return (record).toString().length() + this.OVERHEAD_SIZE_IN_BYTES;
```
Comment (htran1) [inline gobblin-core-base/src/main/java/gobblin/writer/BytesBoundedBatch.java]: Should consider having a record type to support non-string encoding and also avoid the cost of composing the string to get the length.

Comment (yukuai518) [inline gobblin-core-base/src/main/java/gobblin/writer/BytesBoundedBatch.java]: I plan to have another checkin to resolve this issue

**Pull request merged**